sdk/storage/Azure.Storage.Common/src/Shared/ClientsideEncryption/Models/EncryptedBlobRange.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Storage.Cryptography;
+
+namespace Azure.Storage.Blobs.Specialized.Models
+{
+    /// <summary>
+    /// This is a representation of a range of bytes on an encrypted blob, which may be expanded from the requested
+    /// range to include extra data needed for decryption. It contains the original range as well as the calculated
+    /// expanded range.
+    /// </summary>
+    internal struct EncryptedBlobRange
+    {
+        /// <summary>
+        /// The original blob range requested by the user.
+        /// </summary>
+        public HttpRange OriginalRange { get; }
+
+        /// <summary>
+        /// The blob range to actually request from the service that will allow
+        /// decryption of the original range.
+        /// </summary>
+        public HttpRange AdjustedRange { get; }
+
+        public EncryptedBlobRange(HttpRange originalRange)
```

Comment:
I'm not sure if constructor is right place for such heavy operation.
Don't we just need some kind of `HttpRange ExtendRange(HttpRange source)` transformation ?

Reply:
We can. This was ported over from I think Java track 2, so we can change it to suit our needs.

Reply:
Addressed in upcoming push